src/mpl.cpp

```diff
@@ -5,187 +5,194 @@
+    hal::byte updated_reg = reg_buffer[0] | bits_to_set;
+    HAL_CHECK(hal::write(*p_i2c, device_address, std::array<hal::byte, 2>{ reg_addr, updated_reg }, hal::never_timeout()));
 
     return hal::success();
 }
 
-/*
+/**
 * @brief Wait for a specified flag bit in a register to be set to the desired state.
+* @param p_i2c The I2C peripheral used for communication with the device.
 * @param reg: 8 bit value specifying the register address
 * @param flag: 8 bit value specifying which bit(s) to check
 */
-hal::status mpl::poll_flag(hal::byte reg, hal::byte flag, bool desired_state)
+hal::status poll_flag(hal::i2c* p_i2c, hal::byte reg, hal::byte flag, bool desired_state)
 {   
     std::array<hal::byte, 1> status_payload { reg };
     std::array<hal::byte, 1> status_buffer {};
     bool flag_set = true;
 
     while (flag_set) {
-        hal::result<hal::i2c::transaction_t> res = hal::write_then_read(
-                                                            *m_i2c, 
-                                                            device_address, 
-                                                            status_payload, 
-                                                            status_buffer, 
-                                                            hal::never_timeout());
-        // Only update when no errors (catch device not found)
-        if (res) {
+        if (hal::write_then_read(
+                *p_i2c, 
+                device_address, 
+                status_payload, 
+                status_buffer, 
+                hal::never_timeout())
+        ) {
+            // Only update when no errors (catch device not found)
             flag_set = desired_state ? ((status_buffer[0] & flag) == 0) : ((status_buffer[0] & flag) != 0);
         }
     }
 
     return hal::success();
 }
 
-/*
-* @brief Read pressure data from OUT_T_MSB_R and OUT_T_LSB_R
-*   and perform temperature conversion to celsius.
+/**
+* @brief Trigger one-shot measurement by setting ctrl_reg1_ost bit in ctrl_reg1.
+* @param p_i2c The I2C peripheral used for communication with the device.
 */
-hal::result<mpl::temperature_read_t> mpl::t_read()
+hal::status initiate_one_shot(hal::i2c* p_i2c) 
+{
+    // Wait for one-shot flag to clear
+    poll_flag(p_i2c, ctrl_reg1, ctrl_reg1_ost, false);
+
+    // Set ost bit in ctrl_reg1 - initiate one shot measurement
+    HAL_CHECK(modify_reg_bits(p_i2c, ctrl_reg1, ctrl_reg1_ost));
+
+    return hal::success();
+}
+
+/***********************************************
+ *  MPL Class Functions
+ ***********************************************/
+
+mpl::mpl(hal::i2c& p_i2c)
+    : m_i2c(&p_i2c)
+{}
+
+result<mpl> mpl::create(hal::i2c& p_i2c)
+{
+    mpl mpl_dev(p_i2c);
+    
+    // sanity check
+    auto whoami_buffer = HAL_CHECK(hal::write_then_read<1>(p_i2c, device_address, std::array<hal::byte, 1>{ whoami_r }, hal::never_timeout()));
+
+    if (whoami_buffer[0] != 0xC4) {
+        return hal::new_error();
```

Comment:
```suggestion
        return hal::new_error();
```

Just a reminder that this shouldn't return an anonymous error.